src/iotest.f90

```diff
@@ -132,45 +188,103 @@ subroutine iotest
       end do
     end do
 
-    do a = 1,spn
+    if (mag_on) then
       do i = is, ie
         do j = js, je
           do k = ks, ke
             err = 0.d0
-            err = err + abs(spc(a,i,j,k) - (8.d0 + 1.d0*a + 1.d-2*i + 1.d-4*j + 1.d-6*k))
+            err = err + abs(b1(i,j,k) - (5.d0 + 1.d-2*i + 1.d-4*j + 1.d-6*k))
```

Comment:
I think this method of summing up errors is a bit unwieldy (yes, I realise I'm the one who originally implemented it). We could have a function that individually checks each physical quantities, so that we can pinpoint which array the error occurred in. But maybe that can be done in a separate PR.

Reply:
Agreed, we'll fix it up in a new PR